src/coin/skycoin/models/networking.go

```diff
@@ -0,0 +1,114 @@
+package skycoin
+
+import (
+	"log"
+	"strings"
+
+	"github.com/fibercrypto/FiberCryptoWallet/src/core"
+	"github.com/skycoin/skycoin/src/api"
+	"github.com/skycoin/skycoin/src/readable"
+)
+
+type SkycoinNetworkIterator struct {
+	//Implements NetworkIterator interface
+	current  int
+	networks []core.INetwork
+}
+
+func (it *SkycoinNetworkIterator) Value() core.INetwork {
+	return it.networks[it.current]
+}
+
+func (it *SkycoinNetworkIterator) Next() bool {
+	if it.HasNext() {
+		it.current++
+		return true
+	}
+	return false
+}
+
+func (it *SkycoinNetworkIterator) HasNext() bool {
+	if (it.current + 1) >= len(it.networks) {
+		return false
+	}
+	return true
+}
+
+func NewSkycoinNetworkIterator(network []core.INetwork) *SkycoinNetworkIterator {
+	return &SkycoinNetworkIterator{networks: network, current: -1}
+}
+
+type SkycoinRemoteNetwork struct {
+	//Implements NetworkSet interface
+	nodeAddress string
+}
+
+func NewSkycoinRemoteNetwork(nodeAddress string) *SkycoinRemoteNetwork {
+	return &SkycoinRemoteNetwork{nodeAddress}
+}
+
+func (remoteNetwork *SkycoinRemoteNetwork) newClient() *api.Client {
+	return api.NewClient(remoteNetwork.nodeAddress)
+}
+
+func (remoteNetwork *SkycoinRemoteNetwork) ListNetworks() core.NetworkIterator {
```

Comment:
Is this for P2P or API nodes ? P2P peers may not have enabled REST API .

Reply:
in any case `SkycoinRemoteNetwork` is a name I can't understand , should be changed

Reply:
This connect to API nodes, About the name I don't now complete sure about the change I made, I rename it to `SkycoinNetworkConnections`, What do you think?